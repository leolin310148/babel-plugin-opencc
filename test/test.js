const babel = require("@babel/core");
const openccPlugin = require("../index");

var example = `
console.log("正则表达式");
const hello = "hello";
const myFunc = function (param = "判断输入") {
  return "汉字";
};
`;

it("should replace", () => {
  const { code } = babel.transform(example, {
    plugins: [[
      openccPlugin, { translation: "simplifiedToTraditional" }
    ]]
  });
  expect(code).toMatchSnapshot();
});

it("should change nothing", () => {
  const { code } = babel.transform(example, {
    plugins: [[
      openccPlugin, { translation: "traditionalToSimplified" }
    ]]
  });
  expect(code.match(/[\u4e00-\u9fa5]/g)).toEqual(example.match(/[\u4e00-\u9fa5]/g));
});

