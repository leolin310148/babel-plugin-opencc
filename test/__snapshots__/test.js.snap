// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should replace 1`] = `
"console.log(\\"\\\\u6B63\\\\u5247\\\\u8868\\\\u9054\\\\u5F0F\\");
const hello = \\"hello\\";

const myFunc = function (param = \\"\\\\u5224\\\\u65B7\\\\u8F38\\\\u5165\\") {
  return \\"\\\\u6F22\\\\u5B57\\";
};"
`;
